src/ThreadPool.cc

```diff
@@ -134,6 +135,16 @@ std::vector<std::string> ThreadPool::get_running_task_names() const
     return running_task_names_;
 }
 
+std::size_t ThreadPool::get_thread_index() const
+{
+    const std::size_t idx = thread_index_;
+
+    if (idx == std::numeric_limits<std::size_t>::max())
+        throw std::runtime_error("This thread is not part of the pool");
+
+    return idx;
```

Comment:
Why this extra variable? Also I wonder if the index can actually exceed `max_threads` (or `max_capacity` :thinking:) and with that in mind, shouldn't be the check be:
```c++
    if (thread_index_ > max_threads)
        throw std::runtime_error("This thread is not part of the pool");

    return thread_index_;
```
Also I think this method could easily live in the header and therefor been inline-able by the compiler.

Reply:
Good point about the extra variable, I removed that. As for putting the function into the header, I tried that, but somehow g++ does not like it:
```
/usr/bin/ld: tests/libgul-test.p/test_ThreadPool.cc.o: in function `TLS wrapper function for gul14::ThreadPool::thread_index_':
/home/lfroehli/doocs/library/common/gul14/builddir/../include/gul14/ThreadPool.h:539:(.text._ZTWN5gul1410ThreadPool13thread_index_E[_ZTWN5gul1410ThreadPool13thread_index_E]+0x25): undefined reference to `gul14::ThreadPool::thread_index_'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
I guess this is – maybe – a bug in how thread-local storage is implemented in gcc? In any case, inlining is not critical here, so I think we can ignore that.

Reply:
> Also I wonder if the index can actually exceed max_threads

No, it can't because it is assigned exactly once at the start of the thread, and in the current design we cannot shut down threads at runtime (except by terminating the entire ThreadPool).

Reply:
Well of course we wouldn't introduce something that makes it impossible for us to change that in the future.

That would be worse than documenting function return values.